packages/react-composites/src/composites/CallComposite/utils/Utils.ts

```diff
@@ -184,3 +186,70 @@ export const computeVariant = (
     return 'NO_STATE';
   }
 };
+
+/**
+ * Hook to override the avatar persona data of the users
+ * - Calls {@link AvatarPersonaDataCallback} on each render for provided `userIds`
+ * - Returns an array of the same length as `userIds`. Entries in the array may be undefined if there is no data to return.
+ * @private
+ */
+export const useCustomAvatarPersonaData = (
+  userIds: (string | undefined)[],
+  callBack?: AvatarPersonaDataCallback
+): (AvatarPersonaData | undefined)[] => {
+  const [data, setData] = React.useState<(AvatarPersonaData | undefined)[]>([]);
+  useEffect(() => {
+    (async () => {
+      if (!callBack) {
```

Comment:
nit: should be able to avoid this if block,
simply intialize the data to an array of `[undefineds]`
```
const [data, setData] = React.useState<(AvatarPersonaData | undefined)[]>(Array(userIds.length).fill(undefined));
```

Reply:
Nada. That don't work. :D 

Consider the case when `callback` was defined on first render but was then set to `undefined` (Why would Contoso do such a thing? I don't know, but I don't want to have reason globally when reading this function. It _can_ happen).

Reply:
This would then be caught later on under 
```tsx
        userIds.map(async (userId: string | undefined) => {
          if (!userId) {
            return undefined;
          }
          return await callBack(userId);
        })
```

a callback would now be undefined so typescript would complain calling it and this would be
```tsx
userIds.map(async (userId: string | undefined) => {
          if (!userId || !callBack) {
            return undefined;
          }
          return await callBack(userId);
        })
```

Reply:
I snuck onto Donald's machine and made the change, let me know if I missed something haha